internal/stage_magnet_parse.go

```diff
@@ -0,0 +1,58 @@
+package internal
+
+import (
+	"fmt"
+
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+func testParseMagnetLink(stageHarness *test_case_harness.TestCaseHarness) error {
+	type MagnetLink struct {
+		UrlEncoded string
+		Trackers []string
+		InfoHash string
+	}
+
+	magnetLinks := []MagnetLink {
+		{
+		   UrlEncoded: "magnet:?xt=urn:btih:c77829d2a77d6516F88cd7a3de1a26abcbfab0db&dn=codercat.gif&tr=http%3A%2F%2Fbittorrent-test-tracker.codecrafters.io%2Fannounce",
```

Comment:
We need some level of randomization for both the info hash and tracker URL here. One could pass this stage by just hardcoding both - and often people do this just to ensure they have the "format" correct, and then intend to implement the actual parsing after .

Reply:
("Randomization" here could just be including three different cases with different values - don't necessarily mean truly random values)

Reply:
Good point, will update